eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -12,21 +12,47 @@ parameters:
   default: []
 - name: JobTemplatePath
   type: string
+# Set this to false to do a full checkout for private repositories with the azure pipelines service connection
+- name: SparseCheckout
+  type: boolean
+  default: true
+- name: SparseCheckoutPaths
+  type: object
+  default: []
+- name: Pool
+  type: string
+  default: azsdk-pool-mms-ubuntu-1804-general
+- name: OsVmImage
+  type: string
+  default: MMSUbuntu18.04
 
 jobs:
 - job: generate_matrix
   variables:
     displayNameFilter: $[ coalesce(variables.jobMatrixFilter, '.*') ]
   pool:
-    name: Azure Pipelines
-    vmImage: ubuntu-18.04
+    name: ${{ parameters.Pool }}
+    vmImage: ${{ parameters.OsVmImage }}
   displayName: Generate Job Matrix
   steps:
+    # Skip sparse checkout for the `azure-sdk-for-<lang>-pr` private mirrored repositories
+    # as we require the github service connection to be loaded.
+    - ${{ if and(parameters.SparseCheckout, not(contains(variables['Build.DefinitionName'], '-pr - '))) }}:
```

Comment:
I'm not sure if it is better to check against the pipeline or repo name? I have a slight preference for the repo name so we aren't as dependent on someone failing to name their pipeline correctly.

Reply:
Repo name isn't available at template build time :(

Reply:
https://github.com/Azure/azure-sdk-tools/pull/1452#issuecomment-790977989

Reply:
booo... I guess we will have to see how this works out. I don't like how loosely coupled this is.

Reply:
I suspect this is going to come back to bite us once we start using it in more places in the private repos, but for now I guess we can start here and then reconsider other options if we hit an issue.

Reply:
Yeah, I'm not a fan of this approach but can't find an alternative. My thinking is also that the `pr` repos are right now our only standardized name private repos, and people that need to do temporary feature/pre-release testing can just keep `SparseCheckout: false` in their branch until they need to go live.

Reply:
It would be nice if they didn't have to have a code commit that would need to be undone once the merge to public. We could consider a pipeline variable that someone could set to opt-in/out of this but lets see if any when this becomes an issue.

Reply:
The variable would make the template harder to define since we'd have to use a condition. I was using a `SparseCheckout` top-level parameter myself for testing, but if I had to check-enable it every time for private repo testing I think I would end up making a commit anyway as part of my workflow.